src/core/public/application/types.ts

```diff
@@ -639,8 +643,8 @@ export interface InternalApplicationStart
     provider: IContextProvider<AppMountDeprecated, T>
   ): void;
 
-  // Internal APIs
   currentAppId$: Observable<string | undefined>;
```

Comment:
This can be removed and instead add `| 'currentAppId$'` to the `Pick` on line 619 (623 on the new diff)